### DISCUSSION TITLE
How to monitor/health check self-hosted Supabase services?

### DISCUSSION THREAD

**Discussion by dule-kitemetric** [supabase/supabase#31200] [Questions]
# Context
- I'm **self-hosting** Supabase services (auth, db, rest,...) to use in my web application. All authentication & database-querying logics are handled at backend API using `service_role` key
- Currently, I want to monitor my API Application using [Better Stack](https://betterstack.com/) (uptime). My strategy is to create a endpoint `/health/ready` to check its readiness
  -  I want to check if the Supabase `auth` (for authentication), `rest & db` (for database-querying) are still alive so that i'll assume the API application is ready to receive request
# Question
- Does Supabase client have a method to check the current status of Supabase authentication & database-querying capabilities?
  - E.g. for checking database-querying capability, i can use `supabase.from('my_table').select()`.
  If there's no error then it's OK. But this might not be a good idea since i have to query directly to the database's table

Thank you so much!

**Comment by GaryAustin1** [comment]
For what it is worth the Supabase dashboard uses these for database and auth:  
![image](https://github.com/user-attachments/assets/52e79e24-284a-4674-8008-35a4024875e1)
